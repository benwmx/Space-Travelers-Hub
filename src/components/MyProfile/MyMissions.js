import React from 'react';
import { useSelector } from 'react-redux';

const MyMissions = () => {
  const myMissions = useSelector((state) => state.missions.filter((mission) => mission.joined));
  return (
    <div className="myMissionsContainer">
      <ul>
        {
          myMissions.map((mission) => (
            <li key={mission.id}>
              {mission.name}
            </li>
          ))
        }
      </ul>
    </div>
  );
};

export default MyMissions;
