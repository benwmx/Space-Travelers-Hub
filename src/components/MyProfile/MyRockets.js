import React from 'react';

const MyRockets = () => (
  <p> Under construction ğŸ‘·</p>
);

export default MyRockets;
