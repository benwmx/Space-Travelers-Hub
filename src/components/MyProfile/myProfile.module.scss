.container {
  width: 90%;
  margin: 50px auto;
  display: flex;
  justify-content: space-between;

  .listContainer {
    width: 40%;

    h2 {
      color: #e6b800;
    }

    .list {
      border: 1px solid rgb(146, 142, 142);
      background-color: rgba(253, 253, 253, 0.8);
      box-shadow: 5px 5px 10px  rgb(148, 148, 145);
      list-style-type: none;
      padding-left: 0;
      -webkit-border-radius: 6px;
      -moz-border-radius: 6px;
      -ms-border-radius: 6px;
      -o-border-radius: 6px;
      border-radius: 6px;

      li {
        border-bottom: 1px solid rgb(146, 142, 142);
        padding: 10px;
        font-size: 1.5rem;
      }
    }
  }
}
