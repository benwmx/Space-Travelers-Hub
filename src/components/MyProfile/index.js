import React from 'react';
import MyMissions from './MyMissions';
import MyRockets from './MyRockets';

const MyProfile = () => (
  <div className="container">
    <MyMissions />
    <MyRockets />
  </div>
);

export default MyProfile;
