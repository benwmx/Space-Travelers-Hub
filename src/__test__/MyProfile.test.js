import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

describe('MyProfile Component test', () => {
  it('Should render', () => {
    const app = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
