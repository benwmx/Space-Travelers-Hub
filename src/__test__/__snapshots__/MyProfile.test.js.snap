// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component test Should render 1`] = `
<div
  className="container"
>
  <div
    className="listContainer"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="list"
    />
  </div>
  <div
    className="listContainer"
  >
    <h2
      className="d"
    >
      My Missions
    </h2>
    <p>
      Empty ðŸ˜¢ 
    </p>
  </div>
</div>
`;
